File: app/src/main/java/com/exemple/urbanbus/data/api/OlhoVivoAPI.kt
package com.exemple.urbanbus.data.api

import com.exemple.urbanbus.data.dtos.BusPostitionDTO
import com.exemple.urbanbus.data.dtos.BusStopArrivalDTO
import com.exemple.urbanbus.data.models.BusLine
import com.exemple.urbanbus.data.models.BusStop
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Query

// interface para realizacao de request para api
interface OlhoVivoAPI {
    @POST("Login/Autenticar")
    suspend fun authenticate(@Query("token") token: String): Response<String>

    @GET("Parada/Buscar")
    suspend fun getBusStops(
        @Header("Cookie") cookie: String,
        @Query("termosBusca") term: String = ""
    ): List<BusStop>

    @GET("Previsao/Parada")
    suspend fun getStopArrival(
        @Header("Cookie") cookie: String,
        @Query("codigoParada") busStopCode: String = ""
    ): BusStopArrivalDTO

    @GET("Linha/Buscar")
    suspend fun getBusLines(
        @Header("Cookie") cookie: String,
        @Query("termosBusca") busStopCode: String
    ): List<BusLine>

    @GET("Posicao/Linha")
    suspend fun getBusPosition(
        @Header("Cookie") cookie: String,
        @Query("codigoLinha") busStopCode: Int
    ): BusPostitionDTO
}

File: src/services/OlhoVivoAPI.js
import axios from 'axios';

const API_URL = 'http://api.olhovivo.sptrans.com.br/v2.1';

const authenticate = async () => {
  const response = await axios.post(`${API_URL}/Login/Autenticar`, null, {
    params: { token: '9fdeec300d5efd3b8fa3d2ec79b7a59b0af6b3b44947f5594090ea66512530e8' }
  });
  return response.data;
};

const getVehicles = async () => {
  const response = await axios.get(`${API_URL}/Posicao`);
  return response.data;
};

export { authenticate, getVehicles };
